cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/ProgramWorkflowRunnerFactory.java

```diff
@@ -60,17 +60,18 @@ public ProgramWorkflowRunnerFactory(WorkflowSpecification workflowSpec, ProgramR
    * @param token      The {@link WorkflowToken}
    * @return the appropriate concrete implementation of {@link ProgramWorkflowRunner} for the program
    */
-  public ProgramWorkflowRunner getProgramWorkflowRunner(WorkflowActionSpecification actionSpec, WorkflowToken token) {
+  public ProgramWorkflowRunner getProgramWorkflowRunner(WorkflowActionSpecification actionSpec, WorkflowToken token,
```

Comment:
`WorkflowActionSpecification` does not contain the node id. 

The hierarchy is, at the top level we have `WorkflowNode`, which contains `nodeId` and the `nodeType`. Action nodes are represented by the `WorkflowActionNode` class which inherits the `WorkflowNode` and include the `WorkflowActionSpecification` as well. 

Please let me know if nodeId should also be included in the action spec.